test/city-council-district/city-council-district.e2e-spec.ts

```diff
@@ -104,4 +107,107 @@ describe("City Council District e2e", () => {
       expect(response.body.error).toBe(HttpName.INTERNAL_SEVER_ERROR);
     });
   });
+
+  describe("findCapitalProjectsByCityCouncilDistrictId", () => {
+    const mock =
+      cityCouncilDistrictRepositoryMock.checkCityCouncilDistrictByIdMocks[0];
+
+    it("should 200 amd return all capital projects for a city council district", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects`)
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(20);
+      expect(parsedBody.offset).toBe(0);
+    });
+
+    it("should 200 amd return all capital projects for a city council district with user specifyied offset and limit", async () => {
+      const limit = 10;
+      const offset = 3;
+      const response = await request(app.getHttpServer())
+        .get(
+          `/city-council-districts/${mock.id}/capital-projects?limit=${limit}&offset=${offset}`,
+        )
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(limit);
+      expect(parsedBody.offset).toBe(offset);
+    });
+
+    it("should 400 and when finding by an invalid id", async () => {
+      const invalidId = "MN";
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${invalidId}/capital-projects`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an invalid limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=b4d"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an 'too-high' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=25"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+    it("should 400 and when finding by an 'too-low' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=0"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
```

Comment:
I'm actually confused as to why these are 400 errors.

Limits from 0 to 100 (inclusive) should be valid. 20 is the default but 100 is the max

Reply:
Correction:  A limit of '0' is indeed too low and will error. Limits from 1 to 100 (inclusive) are valid. I'm still confused as to why the case of "limit=25" is causing a 400

Reply:
The root cause was a trailing `"` after the `limit` parameter, which makes the param invalid and causing the test to pass with `200` when it should have `400`ed